### PR TITLE
[feature](storage-vault) Implement privilege management for storage vault

### PR DESCRIPTION
**Pull request by platoneko**
## Proposed changes

Implement privilege management for storage vault

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by gavinchou — COMMENTED** [review]
LGTM

**Comment by gavinchou** [comment]
need regression test

**Comment by platoneko** [comment]
run buildall

File: regression-test/suites/vaults/privilege.groovy [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import java.util.stream.Collectors;
+
+suite("test_privilege_vault") {
```
Comment (gavinchou) [inline regression-test/suites/vaults/privilege.groovy:20]: how about setting the default vault?

**Comment by platoneko** [comment]
run buildall

**Comment by platoneko** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by platoneko** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**